libs/natparse/src/main/java/org/amshove/natparse/lexing/Lexer.java

```diff
@@ -843,6 +850,33 @@ private void consumeAsteriskOrSystemVariable()
 		createAndAddCurrentSingleToken(SyntaxKind.ASTERISK);
 	}
 
+	private boolean consumeNumberedLabel()
+	{
+		var previous = previous();
+		if (previous != null && previous.kind() == SyntaxKind.IDENTIFIER)
+		{
+			return false;
+		}
+
+		var i = 1; // Skip '(', then check if next 4 are digits
+		while (!scanner.isAtEnd() && Character.isDigit(scanner.peek(i)) && i < 5)
```

Comment:
This should be `scanner.isAtEnd(i)` to check if the offset `i` is still in bounds

Reply:
good find 👍